addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java

```diff
@@ -323,6 +324,17 @@ protected void addTests(Object testsObj, AutomationProgress progress) {
                     Constant.messages.getString(
                             "automation.tests.stats.nullInMemoryStats", getType()));
         }
+        if (((ArrayList<?>) testsObj)
+                        .stream()
+                                .map(LinkedHashMap.class::cast)
+                                .map(t -> t.get("type"))
+                                .anyMatch("alert"::equals)
+                && Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.class)
```

Comment:
Might be better to put this test in the AutomationAlertTest constructor?
I'd prefer AutomationJob to know as little as possible about the job and test internals.

Reply:
This check is returning (in which case the tests are not added at all). IMO better to refactor how the tests are managed in a follow up PR.